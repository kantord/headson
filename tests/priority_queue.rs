use assert_cmd::Command;
use insta::assert_snapshot;

#[test]
fn pq_empty_array() {
    let mut cmd = Command::cargo_bin("headson").unwrap();
    let output = cmd.arg("-f").arg("json").write_stdin("[]").output().unwrap();
    let stderr = String::from_utf8_lossy(&output.stderr);
    assert_snapshot!("pq_empty_array_stderr", stderr);
}

#[test]
fn pq_single_string_array() {
    let mut cmd = Command::cargo_bin("headson").unwrap();
    let output = cmd.arg("-f").arg("json").write_stdin("[\"ab\"]").output().unwrap();
    let stderr = String::from_utf8_lossy(&output.stderr);
    assert_snapshot!("pq_single_string_array_stderr", stderr);
}
