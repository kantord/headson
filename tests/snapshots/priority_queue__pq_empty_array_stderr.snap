---
source: tests/priority_queue.rs
expression: stderr
---
queue_size=1
